### PR TITLE
Add pre/post hooks to deploy/cleanup script and focus/skip to e2e script

### LINKED ISSUE #508: Prepare for adding e2e tests for specific configuration
To add e2e tests that only works in specific configuration, below features are needed:

- Pre/post hooks in deploy/cleanup script
- Focus/skip in e2e script

Detailed design idea is descirbed [here](https://github.com/submariner-io/submariner/pull/1202#discussion_r602476360) .

### PR DESCRIPTION
**Pull request by mkimuram**
This PR adds below features to shipyard e2e scripts:

- Pre/post hooks in deploy/cleanup script
- Focus/skip in e2e script

Fixes: #508

### CODE REVIEW

**Comment by mkimuram** [comment]
As described in commmit messages, these features can be used:

- Pre/post hooks in deploy/cleanup script:
```
export PLUGIN=/go/src/github.com/submariner-io/shipyard/scripts/shared/resources/plugins/sample_hook
make deploy
make cleanup
```

- Focus/skip in e2e script:
```
export FOCUS="\[example\]"
export SKIP="Should be able to list"
make e2e
```

**Comment by mkimuram** [comment]
@dfarrell07 

I've implemented the scripts. Please review it.

File: gh-actions/e2e/action.yaml [hunk 1]
```diff
@@ -12,6 +12,18 @@ inputs:
     description: 'Working directory to run in'
     required: false
     default: '.'
+  focus:
+    description: '-ginkgo.focus argument to passed to e2e tests'
```
Comment (dfarrell07) [inline gh-actions/e2e/action.yaml]:
```suggestion
    description: '-ginkgo.focus argument to passed to E2E tests'
```

Nit: I think we capitalize it consistently in this context, so let's continue doing so.

File: gh-actions/e2e/action.yaml [hunk 2]
```diff
@@ -12,6 +12,18 @@ inputs:
...
+    required: false
+    default: '\[.*\]'
+  skip:
+    description: '-ginkgo.skip argument to passed to e2e tests'
```
Comment (dfarrell07) [inline gh-actions/e2e/action.yaml]:
```suggestion
    description: '-ginkgo.skip argument to passed to E2E tests'
```

**Review by dfarrell07 — COMMENTED** [review]
This is great @mkimuram, thank you! One very minor nit inline.

**Comment by dfarrell07** [comment]
There are also some commit message linting issues:

```
gitlint --commits origin/devel..HEAD
Commit 9d8033a5fe:
10: B1 Line exceeds max length (100>80): "export PLUGIN=/go/src/github.com/submariner-io/shipyard/scripts/shared/resources/plugins/sample_hook"

Commit aad86c5d2d:
5: B1 Line exceeds max length (86>80): "- The value of `FOCUS` environment variable is added with --focus option to `E2E_ARGS`"
7: B1 Line exceeds max length (84>80): "- The value of `SKIP` environment variable is added with --skip option to `E2E_ARGS`"
```

**Comment by dfarrell07** [comment]
And also some shellcheck linting issues:

```
shellcheck -e SC2154 scripts/shared/lib/*

In scripts/shared/lib/utils line 138:
    type -t $func | grep -q function || return 0
            ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    type -t "$func" | grep -q function || return 0


In scripts/shared/lib/utils line 139:
    eval $@
         ^-- SC2068: Double quote array expansions to avoid re-splitting elements.

For more information:
  https://www.shellcheck.net/wiki/SC2068 -- Double quote array expansions to ...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

**Comment by mkimuram** [comment]
@dfarrell07 

Thank you for your review. Fixed lint and commit issues and changed e2e to E2E.
It seems that the upgrade test is failing. I'm investigating it.

**Comment by dfarrell07** [comment]
> It seems that the upgrade test is failing. I'm investigating it.

@mkimuram It's not related to your PR, you can ignore it.

**Review by dfarrell07 — APPROVED** [review]
Thanks for the quality PR @mkimuram! LGTM :+1:

File: gh-actions/e2e/action.yaml [hunk 3]
```diff
@@ -12,6 +12,18 @@ inputs:
     description: 'Working directory to run in'
     required: false
     default: '.'
+  focus:
+    description: '-ginkgo.focus argument to passed to E2E tests'
```
Comment (tpantelis) [inline gh-actions/e2e/action.yaml]:
```suggestion
    description: '-ginkgo.focus argument to pass to E2E tests'
```

File: gh-actions/e2e/action.yaml [hunk 4]
```diff
@@ -12,6 +12,18 @@ inputs:
...
+    required: false
+    default: '\[.*\]'
+  skip:
+    description: '-ginkgo.skip argument to passed to E2E tests'
```
Comment (tpantelis) [inline gh-actions/e2e/action.yaml]:
```suggestion
    description: '-ginkgo.skip argument to pass to E2E tests'
```

**Review by tpantelis — CHANGES_REQUESTED** [review]
Overall looks good.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**